Italicise "bytes" in the docs of some `Vec` methods

gurry opened the pull request:
On a cursory read it's easy to miss that the limit is in terms of bytes not no. of elements. The italics should help with that.

Fixes #119149

bsilver8192 commented:
LGTM. I think this would've caught my attention before filing the issue in question.

gurry committed:
Italicise "bytes" in the docs of some `Vec` methods

because on a cursory read it's easy to miss that the limit is
in terms of bytes not no. of elements. The italics should help
with that.

cuviper commented:
Sounds good!

@bors r+ rollup

bors commented:
:pushpin: Commit e3aca01343a06b3871b3b0f920c91f95be54a6d9 has been approved by `cuviper`

It is now in the [queue](https://bors.rust-lang.org/homu/queue/rust) for this repository.

<!-- @bors r=cuviper e3aca01343a06b3871b3b0f920c91f95be54a6d9 -->
<!-- homu: {"type":"Approved","sha":"e3aca01343a06b3871b3b0f920c91f95be54a6d9","approver":"cuviper","queue":"https://bors.rust-lang.org/homu/queue/rust"} -->
